codespell_lib/_codespell.py

```diff
@@ -567,35 +556,34 @@ def _script_main():
 
 def main(*args):
     """Contains flow control"""
-    global options
-    global quiet_level
-    global file_opener
-
     options, args, parser = parse_options(args)
 
-    if options.regex:
-        if options.write_changes:
-            parser.error('--write-changes cannot be used together with '
-                         '--regex')
-        global word_regex
-        try:
-            word_regex = re.compile(options.regex)
-        except re.error as err:
-            print('ERROR: invalid regular expression "%s" (%s)' %
-                  (options.regex, err), file=sys.stderr)
-            parser.print_help()
-            return 1
+    if options.regex and options.write_changes:
+        print('ERROR: --write-changes cannot be used together with '
```

Comment:
Out of interest why have you switched away from .error?

Reply:
Ahh yes just missed that it was a `parser.error` before

Reply:
... actually I realized this is a consistency thing. Everywhere else we `print('ERROR: ...); return 1` for errors.

Reply:
I wonder if it might be good to do different exit codes for different error states in future, but that's probably for another day!